File: hello_world/hello_world.py
from pyspark import SparkContext, SparkConf

conf = SparkConf().setAppName("Word Counter")
sc = SparkContext(conf=conf)

text_file = sc.textFile("file:///opt/spark/README.md")
tokenized_file_data = text_file.flatMap(lambda line: line.split(" "))
count_prep = tokenized_file_data.map(lambda word: (word, 1))
counts = count_prep.reduceByKey(lambda accum_val, new_val: accum_val + new_val)
sorted_counts = counts.sortBy(lambda kv_pair: kv_pair[1], ascending=False)
sorted_counts.saveAsTextFile("readmeWordCount")

File: hello_world/hello_world.scala
// this script runs the "hello world" of spark, which is counting words in a document
// run the script with the command `spark-shell -i hello_world.scala`
// this sets up the spark context automatically in the scala REPL
// going forward I'm going to try and do everything in python/pyspark, but it is good to know this works in scala too

val textFile = sc.textFile("file:///opt/spark/README.md")
val tokenizedFileData = textFile.flatMap(line=>line.split(" "))
val countPrep = tokenizedFileData.map(word=>(word, 1))
val counts = countPrep.reduceByKey((accumValue, newValue)=>accumValue + newValue)
val sortedCounts = counts.sortBy(kvPair=>kvPair._2, false)
sortedCounts.saveAsTextFile("readmeWordCount")
System.exit(0)